[BUG]: pybind11::python_headers added multiple times when using multi-config cmake generators

### Required prerequisites

- [X] Make sure you've read the [documentation](https://pybind11.readthedocs.io). Your issue may be addressed there.
- [X] Search the [issue tracker](https://github.com/pybind/pybind11/issues) and [Discussions](https:/pybind/pybind11/discussions) to verify that this hasn't already been reported. +1 or comment there if it has.
- [ ] Consider asking first in the [Gitter chat room](https://gitter.im/pybind/Lobby) or in a [Discussion](https:/pybind/pybind11/discussions/new).

### Problem description

When using a multi-config generator like the Visual Studio ones, I get errors on a cmake configure like:

[cmake] CMake Error at C:/Users/tim/.conan/data/pybind11/2.7.1/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/lib/cmake/pybind11/pybind11NewTools.cmake:139 (add_library):
[cmake]   add_library cannot create imported target "pybind11::python_headers"
[cmake]   because another target with the same name already exists.
[cmake] Call Stack (most recent call first):
[cmake]   C:/Users/tim/.conan/data/pybind11/2.7.1/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/lib/cmake/pybind11/pybind11Common.cmake:200 (include)
[cmake]   build/pybind11-config.cmake:410 (include)
[cmake]   CMakeLists.txt:976 (find_package)

This does not happen with single config generators like ninja or make.

It appears to fix the issue if we wrap the add_library in pybind11NewTools.cmake in an if.  In other words, change:

  add_library(pybind11::python_headers INTERFACE IMPORTED)

to:

  if(NOT TARGET pybind11::python_headers)
      add_library(pybind11::python_headers INTERFACE IMPORTED)
  endif()


### Reproducible example code

_No response_

--- a/tools/pybind11Common.cmake
+++ b/tools/pybind11Common.cmake
@@ -20,6 +20,7 @@ Adds the following functions::
 #]======================================================]
 
 # CMake 3.10 has an include_guard command, but we can't use that yet
+# include_guard(global) (pre-CMake 3.10)
 if(TARGET pybind11::lto)
   return()
 endif()

--- a/tools/pybind11NewTools.cmake
+++ b/tools/pybind11NewTools.cmake
@@ -5,6 +5,12 @@
 # All rights reserved. Use of this source code is governed by a
 # BSD-style license that can be found in the LICENSE file.
 
+if(CMAKE_VERSION VERSION_LESS 3.12)
+  message(FATAL_ERROR "You cannot use the new FindPython module with CMake < 3.12")
+endif()
+
+include_guard(GLOBAL)
+
 get_property(
   is_config
   TARGET pybind11::headers
@@ -14,10 +20,6 @@ if(pybind11_FIND_QUIETLY)
   set(_pybind11_quiet QUIET)
 else()
   set(_pybind11_quiet "")
-endif()
-
-if(CMAKE_VERSION VERSION_LESS 3.12)
-  message(FATAL_ERROR "You cannot use the new FindPython module with CMake < 3.12")
 endif()
 
 if(NOT Python_FOUND

--- a/tools/pybind11Tools.cmake
+++ b/tools/pybind11Tools.cmake
@@ -5,6 +5,11 @@
 # All rights reserved. Use of this source code is governed by a
 # BSD-style license that can be found in the LICENSE file.
 
+# include_guard(global) (pre-CMake 3.10)
+if(TARGET pybind11::python_headers)
+  return()
+endif()
+
 # Built-in in CMake 3.5+
 include(CMakeParseArguments)
 

